src/dune_util/dune_flock.c

```diff
@@ -0,0 +1,49 @@
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+
+#ifndef _WIN32
+#include <sys/file.h>
+#endif
+
+#define FD_val(value) Int_val(value)
+
+CAMLprim value dune_flock_lock(value v_fd, value v_block, value v_exclusive) {
+#ifdef _WIN32
+  caml_failwith("no flock on win32");
```

Comment:
There's `LockFileEx` on Windows https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-lockfileex which seems pretty similar to `flock`. Perhaps it can be used to implement this C stub on Windows with similar semantics which would allow using a single codepath in the OCaml side.

Reply:
That would be useful indeed.

Reply:
See #6523